fps/src/main/scala/chapter03/exercise3_24.scala

```diff
@@ -0,0 +1,48 @@
+
+package chapter03
+
+import chapter03.listImplementation.*
+import chapter03.listImplementation.List.*
+
+import scala.annotation.tailrec
+
+
+object exercise3_24 extends App:
+
+ 
+  def hasSubSequence[A](sup: List[A], sub: List[A]): Boolean =
+
+    def innerloop[A](supsup: List[A],checks: List[A] = List.empty): Boolean =
```

Comment:
you can scope local functions like this:

```
def outer(...): Boolean =
  def inner(...): Boolean = ???
  ???
```

( `...`s and `???`s left as an exercise for the reader ;))